org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/TargetNamespace_1CodeAction.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2020 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * Red Hat Inc. - initial API and implementation
+ */
+package org.eclipse.lemminx.extensions.contentmodel.participants.codeactions;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import org.eclipse.lemminx.commons.CodeActionFactory;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.services.extensions.ICodeActionParticipant;
+import org.eclipse.lemminx.services.extensions.IComponentProvider;
+import org.eclipse.lemminx.settings.EnforceQuoteStyle;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Range;
+
+/**
+ * CodeAction to replace an incorrect namespace in an .xml document.
+ * 
+ * Changes the value of the xmlns attribute of the root element of the .xml
+ * document to the declared namespace of the referenced .xsd document.
+ */
+public class TargetNamespace_1CodeAction implements ICodeActionParticipant {
+
+	private static final Pattern NAMESPACE_EXTRACTOR = Pattern.compile("'([^']+)'\\.");
+
+	@Override
+	public void doCodeAction(Diagnostic diagnostic, Range range, DOMDocument document, List<CodeAction> codeActions,
+			SharedSettings sharedSettings, IComponentProvider componentProvider) {
+
+		String namespace = extractNamespace(diagnostic.getMessage());
+		String quote = "\"";
+		if (sharedSettings.getFormattingSettings().getEnforceQuoteStyle() == EnforceQuoteStyle.preferred) {
+			quote = sharedSettings.getPreferences().getQuotationAsString();
+		}
+		// @formatter:off
+		CodeAction replaceNamespace = CodeActionFactory.replace(
+				"Replace with '" + namespace + "'",
+				diagnostic.getRange(),
+				quote + namespace + quote,
+				document.getTextDocument(), diagnostic);
+		// @formatter:on
+		codeActions.add(replaceNamespace);
+	}
+
+	private static String extractNamespace(String diagnosticMessage) {
+		// The error message has this form:
+		// TargetNamespace.1: Expecting namespace 'NaN', but the target namespace of the
+		// schema document is 'http://two-letter-name'.
+		Matcher nsMatcher = NAMESPACE_EXTRACTOR.matcher(diagnosticMessage);
```

Comment:
My worry is what if the namespace contains a `'` (not url encoded like %XX) character?

@angelozerr do we need to worry about this?

Reply:
Good catch. I looked at the specification to see if `'` is allowed in a namespace: https://www.w3.org/TR/xml/#NT-Name 
My understanding from this document is that it isn't.

Reply:
Xerces doesn't complain when you write targetNamespace="'", so I think we should take care of even if it's not a common usecases.

Reply:
Working on it and unit tests for it

Reply:
I implemented it so that it will work with `'` and escapes it when it occurs, but it doesn't recognize the escaped version of `'`, so it still raises `TargetNamespace.1`:

![DoesntFixErrorQuotes](https://user-images.githubusercontent.com/22376627/84042627-4f00b100-a973-11ea-99ed-d73b7e436bcc.gif)

Reply:
I've pushed the code that implements the above (with tests) into this PR. Right now, it always tries to escape single quotes. I left it as a separate commit in case it we want to revert it. Other options could be:

- Always use double quotes when declaring `xmlns` if there is a single quote embedded in the `targetNamespace`
- Don't give a code action if single quotes are prefered and there are single quotes in the `targetNamespace`

Reply:
^This has been reverted back to the code that doesn't deal with single quotes in the namespace